pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -115,10 +115,10 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
 		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
-		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
+		if defErr != nil && !IsConfigurationInvalid(defErr) {
 			return mergedConfig, nil
```

Comment:
@smarterclayton I am still not convinced this is right. Why should we fallback to ICC when the default config has an error, but is not an valid error? Why does default config's error status has such an authoritative bearing on what config is used?

If someone is running a non-host client inside a pod, they populate `mergedConfig` with the config of their API server. `DefaultClientConfig` is never touched. In that case, just because `DefaultClientConfig.ClientConfig()` returned an invalid config, we can't fallback to ICC.